Multisig/UI/Transactions/TransactionDetails/TransactionDetailsViewModel.swift

```diff
@@ -11,57 +11,74 @@ import SwiftUI
-        }
+        reloadData()
     }
 
     init(hash: String) {
-        self.transaction = nil
         self.hash = Data(hex: hash)
+
         super.init()
         reloadData()
     }
 
     enum Failure: LocalizedError {
-        case safeInfoMissing, unsupportedTransaction
-
+        case transactionDetailsNotFound, unsupportedTransaction
         var errorDescription: String? {
             switch self {
-            case .safeInfoMissing:
-                return "No more info available for this transaction"
+            case .transactionDetailsNotFound:
+                return "Information about this transaction can't be loaded"
             case .unsupportedTransaction:
                 return "Information about this transaction type is not supported"
             }
         }
     }
 
     override func reload() {
-        guard isRefreshingEnabled else { return }
-        Just(hash)
-            .compactMap { $0 }
-            .receive(on: DispatchQueue.global())
-            .tryMap { hash -> TransactionViewModel in
-                let transaction = try App.shared.safeTransactionService.transaction(hash: hash)
-                guard let safe = transaction.safe?.address else {
-                    throw Failure.safeInfoMissing
+        if let hash = hash {
+            Just(hash)
+                .compactMap { $0 }
+                .receive(on: DispatchQueue.global())
+                .tryMap { hash -> TransactionViewModel in
+                    let transaction = try App.shared.clientGatewayService.transactionDetails(safeTxHash: hash)
+                    let viewModels = TransactionViewModel.create(from: transaction)
+                    guard viewModels.count == 1, let viewModel = viewModels.first else {
+                        throw Failure.unsupportedTransaction
+                    }
+                    return viewModel
                 }
-                let safeInfo = try App.shared.safeTransactionService.safeInfo(at: safe)
-                let models = TransactionViewModel.create(from: transaction, safeInfo)
-                guard models.count == 1, let model = models.first else {
+                .receive(on: RunLoop.main)
+                .sink(receiveCompletion: { [weak self] completion in
+                    guard let `self` = self else { return }
+                    if case .failure(let error) = completion {
+                        self.errorMessage = error.localizedDescription
+                        App.shared.snackbar.show(message: error.localizedDescription)
+                    }
+                    self.isLoading = false
+                    self.isRefreshing = false
+                }, receiveValue:{ [weak self] transaction in
+                    guard let `self` = self else { return }
+                    self.transactionDetails = transaction
+                    self.errorMessage = nil
+                })
+                .store(in: &subscribers)
+        } else if let id = id {
+            Just(id)
```

Comment:
These two branches are very look-alike, please extract the copy-pasted part into a separate method that would accept a closure parameter that you can execute inside the tryMap